docs/upgrading_to_project_factory_v4.0.md

```diff
@@ -0,0 +1,208 @@
+# Upgrading to Project Factory v4.0
+
+The v4.0 release of Project Factory is a backwards incompatible release and
+features minor changes, specifically with how Google Project Services (APIs)
+are managed. A state migration script is provided to update your existing states
+to minimize or eliminate resource re-creations.
+
+Note that upgrading requires you to have Python 3.7 installed.
+
+## Migration Instructions
+
+This migration was performed with the following example configuration.
+
+```hcl
+/// @file main.tf
+
+provider "google" {
+  credentials = "${file(var.credentials_path)}"
+}
+
+module "my-project" {
+  source            = "terraform-google-modules/project-factory/google"
+  version           = "v3.0.0"
+  random_project_id = "true"
+  name              = "pf-project-services-migrate"
+  org_id            = "${var.org_id}"
+  billing_account   = "${var.billing_account}"
+  credentials_path  = "${var.credentials_path}"
+  activate_apis = [
+    "compute.googleapis.com",
+    "iamcredentials.googleapis.com",
+  ]
+}
+```
+
+### Update the project-factory source
+
+Update the project-factory module source to the Project Factory v4.0.0 release.
+
+```diff
+diff --git i/main.tf w/main.tf
+index d876954..ebb3b1e 100755
+--- i/main.tf
++++ w/main.tf
+@@ -5,7 +5,7 @@ provider "google" {
+
+ module "my-project" {
+   source            = "terraform-google-modules/project-factory/google"
+-  version           = "v3.0.0"
++  version           = "v4.0.0"
+   random_project_id = "true"
+   name              = "pf-project-services-migrate"
+   org_id            = "${var.org_id}"
+```
+
+### Download the state migration script
+
+ ```
+curl -O https://raw.githubusercontent.com/terraform-google-modules/terraform-google-project-factory/v4.0.0/helpers/migrate4.py
+chmod +x migrate4.py
+```
+
+### Reinitialize Terraform
```

Comment:
FYI I don't think the state download is needed for Terraform 0.12. Can probably skip straight to running the script.